current_spring2026/BGE_hybrid_retrieval/reranking.py

```
@@ -0,0 +1,173 @@
+def _extract_years(query: str) -> List[str]:
+    return re.findall(r"\b(1[5-9]\d{2}|20\d{2}|21\d{2})\b", query)
+
+
+def _merge_chunks(docs: List[Document]) -> List[Document]:
+    """Merge all chunks belonging to the same document_id."""
+    grouped: dict = defaultdict(list)
+    for doc in docs:
+        grouped[doc.metadata.get("source")].append(doc)
+
+    merged = []
+    for src, chunks in grouped.items():
```

Ruff (0.15.7)

[warning] Loop control variable `src` not used within loop body

Rename unused `src` to `_src`

(B007)